client/src/containers/PostDashboard.tsx

```diff
@@ -1,13 +1,49 @@
-import React from "react";
-import Button from "@mui/material/Button";
-import Grid from "@mui/material/Grid";
-import Box from "@mui/material/Box";
-import Container from "@mui/material/Container";
-import Typography from "@mui/material/Typography";
-import Pagination from "@mui/material/Pagination";
+import React from 'react';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import Container from '@mui/material/Container';
+import Typography from '@mui/material/Typography';
+import Pagination from '@mui/material/Pagination';
 
-import Header from "../components/Header";
-import PostPreview from "../components/PostPreview";
+import Header from '../components/Header';
+import PostPreview from '../components/PostPreview';
+
+import { api_v1 } from '../api/v1';
+import CreatePost from '../components/CreatePost';
+
+
+function RecentPosts() {
+  const [recentPosts, updateRecent] = React.useState([]);
+
+  const fetchRecentPosts = (limit: number, offset: number) => {
+    api_v1
+      .get('/posts/', {
+        params: { limit, offset },
+      })
+      .then((res: any) => {
+        if (res.data.data.result) {
+          updateRecent(res.data.data.result);
+        }
+      })
+      .catch((err) => console.log(err));
+  };
+
+  React.useEffect(() => {
```

Comment:
We could store the posts in app wide state, set the polling logic in app.tsx, then this way we will always be polling not just when the user is on the posts dashboard.

Reply:
The site does always poll. Since "View post" and "New post" are in dialogs, post dashboard is technically open in the background.